File: PROGRAMMERS/LEVEL2/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> tmp;
    int cnt = 0;
    for (int i = 0; i < progresses.size(); i++) {
        int num = (99 - progresses[i]) / speeds[i] + 1;
        tmp.push_back(num);
    }
    int j;
    for (int i = 0; i < tmp.size(); i++) {
        cnt = 0;
        for (j = i; j < tmp.size(); j++) {
            if (tmp[i] >= tmp[j]) {
                cnt++;
            }
            else {
                i = j - 1;
                break;
            }
        }
        answer.push_back(cnt);
        if (cnt == tmp.size() || j == tmp.size()) break;
    }
    return answer;
}

File: PROGRAMMERS/LEVEL2/기능개발.java
import java.util.*;

class Solution {
    public Queue<Integer> solution(int[] progresses, int[] speeds) {
        
        Queue<Integer> tmp = new LinkedList<Integer>();
        Queue<Integer> answer = new LinkedList<Integer>();
        
        for(int i=0;i<progresses.length;i++) {
        	tmp.add((99-progresses[i])/speeds[i] + 1);
        }
        
        int cnt=1;
        int temp = tmp.poll();
        while(!tmp.isEmpty()) {
        	if(temp>=tmp.peek()) { 
                tmp.poll(); 
                cnt++;
            }
        	else {
        		answer.add(cnt);
        		temp = tmp.poll();
        		cnt=1;
        	}
        }
        answer.add(cnt);
        
        return answer;
    }
}